databridge.camel-aas/src/main/java/org/eclipse/digitaltwin/basyx/databridge/aas/AASPollingConsumer.java

```diff
@@ -143,17 +152,37 @@ private Exchange createExchange(String exchangeProperty) {
 	 * Get serialized metamodel
 	 * @return serialized meta model
 	 */
-	private String getSerializedMetamodel() {
-		
-		if (!getEndpoint().getPropertyPath().isEmpty()) {
-			ConnectedProperty prop = new ConnectedProperty(this.proxy);
-			return new GSONTools(new DefaultTypeFactory()).serialize(prop.getLocalCopy());
+	private String getSerializedMetamodel() throws IOException {
+		if(this.endpoint.getApi() == ApiType.BASYX) {
+			if (!getEndpoint().getPropertyPath().isEmpty()) {
+				ConnectedProperty prop = new ConnectedProperty(this.proxy);
+				return new GSONTools(new DefaultTypeFactory()).serialize(prop.getLocalCopy());
+			}
+
+			ConnectedSubmodel sm = new ConnectedSubmodel(this.proxy);
+			return new GSONTools(new DefaultTypeFactory()).serialize(SubmodelElementMapCollectionConverter.smToMap(sm.getLocalCopy()));
+		}else{
+			String metamodelUrl = getMetamodelUrl().substring(2);
+			return getResponseAsString(executeGetOnURL(metamodelUrl));
 		}
-    	
-		ConnectedSubmodel sm = new ConnectedSubmodel(this.proxy);
-		return new GSONTools(new DefaultTypeFactory()).serialize(SubmodelElementMapCollectionConverter.smToMap(sm.getLocalCopy()));
 	}
-	
+
+	public static String getResponseAsString(CloseableHttpResponse retrievalResponse) throws IOException, ParseException {
+		return EntityUtils.toString(retrievalResponse.getEntity(), "UTF-8");
```

Comment:
As this is just a single line method, we can do this inline and then remove this method. What do you think?